Reference documentation for the Window Controls Overlay API (#11368)
mdn/content: files/en-us/web/api/windowcontrolsoverlaygeometrychangeevent/windowcontrolsoverlaygeometrychangeevent/index.md

---
title: WindowControlsOverlayGeometryChangeEvent()
slug: Web/API/WindowControlsOverlayGeometryChangeEvent/WindowControlsOverlayGeometryChangeEvent
tags:
  - API
  - Constructor
  - DOM Events
  - WindowControlsOverlayGeometryChangeEvent
browser-compat: api.WindowControlsOverlayGeometryChangeEvent.WindowControlsOverlayGeometryChangeEvent
---
{{APIRef("DOM Events")}}

The **`WindowControlsOverlayGeometryChangeEvent()`** constructor returns a newly created
{{domxref("WindowControlsOverlayGeometryChangeEvent")}}, representing the current geometry of a desktop Progressive Web App's title bar area.

## Syntax

```js
new WindowControlsOverlayGeometryChangeEvent(type, {visible: aBooleanValue, titlebarAreaRect: aDOMRectValue});
```

### Arguments

_The `WindowControlsOverlayGeometryChangeEvent()` constructor also inherits arguments from
{{domxref("Event.Event", "Event()")}}._

- _type_

>   - : Is a {{domxref("DOMString")}} representing the name of the type of the

Reviewer: ```suggestion
  - : A {{domxref("DOMString")}} indicating the event type, always 
```

Author: I'm not sure I follow what you mean by `always`?
The value of this string is `change` when the event is fired. I don't know if any other string values are permitted when constructing the event though.

Author: I ended up removing the rest of the sentence, and only keeping `A {{domxref("DOMString")}} indicating the event type.` Is that ok?

Reviewer: What I'm trying to convey is that the exact type name of the event must be supplied in that parameter, and that that name is case sensitive. I also want to save readers from guessing. The spec doesn't make this clear because spec implementors are expected to know the idiom.

    `WindowControlsOverlayGeometryChangeEvent`. It is case-sensitive.

> - `visible`
>   - : Is a boolean flag that's true when the titleBarAreaRect's values are not 0.
> - `titleBarAreaRect`
>   - : Is a {{domxref("DOMRect")}} representing the position and size of the title bar area.

Reviewer: ```suggestion
- `options`
  - : An object with the following properties:
  - `visible`
    - : A boolean flag that's true when the `titleBarAreaRect` object's values are not 0.
  - `titleBarAreaRect`
    - : A {{domxref("DOMRect")}} representing the position and size of the title bar area.
```
